src/openzaak/components/catalogi/admin/mixins.py

```diff
@@ -63,17 +63,25 @@ def response_post_save_change(self, request, obj):
             storage = messages.get_messages(request)
             for i in storage:
                 pass
+            # clear for change view
+            if len(storage._loaded_messages) == 1:
+                del storage._loaded_messages[0]
 
             errors = self._publish_validation_errors(obj)
             if errors:
                 for error in errors:
                     self.message_user(request, error, level=messages.ERROR)
             else:
-                obj.publish()
-                msg = _("The resource has been published successfully!")
-                self.message_user(request, msg, level=messages.SUCCESS)
+                try:
+                    obj.publish()
+                    msg = _("The resource has been published successfully!")
+                    self.message_user(request, msg, level=messages.SUCCESS)
+                    return HttpResponseRedirect(request.path)
+                except ValidationError as e:
```

Comment:
On the second thought I think that there is a special place designed for publishing errors - `_publish_validation_errors` method, which is called above. Could you add this check there? It would be more readable

Reply:
Check moved there